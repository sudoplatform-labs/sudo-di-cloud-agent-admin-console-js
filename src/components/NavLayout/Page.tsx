import styled from 'styled-components';
import { theme } from '../../theme';

export const PageHeading = styled.h1`
  font-size: 24px;
  font-weight: bold;
  color: ${theme.greys.woodsmoke};
`;

export const Page = styled.div`
  padding: 20px 20px 60px 20px;
  max-width: 1000px;
  width: 100%;
`;

export const StretchPage = styled.div`
  flex: 1;
  display: flex;
  width: 100%;
`;
