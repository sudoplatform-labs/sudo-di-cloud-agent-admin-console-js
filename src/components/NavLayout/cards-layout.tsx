import { Col, Row } from 'antd';
import React from 'react';
import styled from 'styled-components';

export const CardsRow: React.FC = (props) => (
  <Row gutter={[16, 32]}>{props.children}</Row>
);

export const CardsCol = styled(Col)`
  > * {
    height: 100%;
  }
`;
