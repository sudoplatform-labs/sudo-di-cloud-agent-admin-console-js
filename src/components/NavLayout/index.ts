export { NavLayout } from './NavLayout';
export * from './Page';
export * from './cards-layout';
