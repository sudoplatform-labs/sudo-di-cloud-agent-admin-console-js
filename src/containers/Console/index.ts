export * from './Console';
export * from './routes';
