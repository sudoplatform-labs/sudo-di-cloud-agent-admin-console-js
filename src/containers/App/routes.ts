import { createRouteRedirect } from '../../components/RouteRenderer';
import { RouteDefinitionFactory } from '../../routes';
import { Console } from '../Console';

export const getRoot: RouteDefinitionFactory = () => ({
  path: '/',
  component: createRouteRedirect('/console'),
  routes: [
    {
      path: '/console',
      component: Console,
      exact: false,
    },
  ],
});
