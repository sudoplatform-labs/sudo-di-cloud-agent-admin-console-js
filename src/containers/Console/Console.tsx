import React from 'react';
import { NavLayout } from '../../components/NavLayout';
import { RouteRenderer } from '../../components/RouteRenderer';
import { getConsoleRoutesRoot } from './routes';

/**
 * <Console> renders the main app navigation, and console routes deal
 * with customer-level functionality.
 */
export const Console: React.FC = () => {
  return (
    <NavLayout>
      <RouteRenderer routeFactory={getConsoleRoutesRoot} />
    </NavLayout>
  );
};
