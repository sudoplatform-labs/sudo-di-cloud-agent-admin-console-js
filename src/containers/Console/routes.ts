import { RouteDefinitionFactory } from '../../routes';
import * as Pages from '../../pages';
import { createRouteRedirect } from '../../components/RouteRenderer';

export const getConsoleRoutesRoot: RouteDefinitionFactory = () => ({
  path: '/console',
  component: createRouteRedirect('/console/dashboard'),
  routes: [
    {
      path: '/console/dashboard',
      component: Pages.ConsoleDashboard,
      displayName: 'Cloud Agent Dashboard',
      iconName: 'home',
      iconClass: 'colored-icon',
      showInSidebar: true,
    },
    {
      path: '/console/settings',
      component: Pages.Settings,
      displayName: 'Settings',
      iconName: 'settings',
      iconClass: 'colored-icon',
      showInSidebar: true,
    },
    {
      path: '/console/decentralized-identifiers',
      component: Pages.DecentralizedIdentifiersDashboard,
      displayName: 'Decentralized Identifiers',
      iconName: 'icon-did-line',
      iconClass: 'colored-icon',
      showInSidebar: true,
    },
    {
      path: '/console/connections',
      component: Pages.ConnectionsDashboard,
      displayName: 'Connections',
      iconName: 'icon-connection-line',
      iconClass: 'colored-icon',
      showInSidebar: true,
    },
    {
      path: '/console/holder-wallet',
      component: createRouteRedirect('/console/holder-wallet/credentials'),
      displayName: 'Holder Wallet',
      iconName: 'icon-wallet-line',
      iconClass: 'colored-icon',
      showInSidebar: true,
      routes: [
        {
          path: '/console/holder-wallet/credentials',
          component: Pages.Credentials,
          displayName: 'My Credentials',
          showInSidebar: true,
        },
        {
          path: '/console/holder-wallet/proofs',
          component: Pages.PresentedProofs,
          displayName: 'My Proof Presentations',
          showInSidebar: true,
        },
      ],
    },
    {
      path: '/console/credential-issuer',
      component: createRouteRedirect('/console/credential-issuer/dashboard'),
      displayName: 'Credential Issuer',
      iconName: 'icon-issuer-line',
      iconClass: 'colored-icon',
      showInSidebar: true,
      routes: [
        {
          path: '/console/credential-issuer/dashboard',
          component: Pages.CredentialIssuerDashboard,
          displayName: 'Dashboard',
          showInSidebar: true,
        },
        {
          path: '/console/credential-issuer/credential-catalogue',
          component: Pages.CredentialCatalogue,
          displayName: 'Catalogue',
          showInSidebar: true,
        },
        {
          path: '/console/credential-issuer/credential-requests',
          component: Pages.CredentialIssuance,
          displayName: 'Credential Issuance',
          showInSidebar: true,
        },
      ],
    },
    {
      path: '/console/credential-verifier',
      component: createRouteRedirect('/console/credential-verifier/dashboard'),
      displayName: 'Credential Verifier',
      iconName: 'icon-verifier-line',
      iconClass: 'colored-icon',
      showInSidebar: true,
      routes: [
        {
          path: '/console/credential-verifier/dashboard',
          component: Pages.CredentialVerifierDashboard,
          displayName: 'Dashboard',
          showInSidebar: true,
        },
        {
          path: '/console/credential-verifier/proofs',
          component: Pages.RequestedProofs,
          displayName: 'Proof Requests',
          showInSidebar: true,
        },
      ],
    },
  ],
});
