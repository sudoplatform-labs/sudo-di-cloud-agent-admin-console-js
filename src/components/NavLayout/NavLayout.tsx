import { Layout } from 'antd';
import React, { useMemo } from 'react';
import { useLocation } from 'react-router';
import styled from 'styled-components';
import { getConsoleRoutesRoot } from '../../containers/Console/routes';
import { flattenRoutes } from '../../routes/utils';
import { BreadCrumbs } from '../BreadCrumbs';
import { AppHeader as Header } from '../Header';
import { Sidebar } from '../Sidebar';

const Container = styled(Layout)`
  height: 100%;
  width: 100%;
`;

const StyledLayoutHeader = styled(Layout.Header)`
  padding: 0;
`;

const StyledLayoutContent = styled(Layout.Content)`
  flex: 1;
  display: flex;
  flex-direction: column;
`;

const StyledLayoutSider = styled(Layout.Sider)``;

const MainContent = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const NavLayout: React.FC = (props) => {
  const location = useLocation();

  const allRoutesFlat = useMemo(
    () => flattenRoutes([getConsoleRoutesRoot()]),
    [],
  );

  const sidebarRoutes = useMemo(() => getConsoleRoutesRoot().routes ?? [], []);

  return (
    <Container>
      <StyledLayoutSider width="252px">
        <Sidebar routes={sidebarRoutes} pathName={location.pathname} />
      </StyledLayoutSider>
      <Layout>
        <StyledLayoutHeader>
          <Header />
        </StyledLayoutHeader>
        <StyledLayoutContent id="STYLED_LAYOUT_CONTENT">
          <BreadCrumbs routes={allRoutesFlat} urlPath={location.pathname} />
          <MainContent id="TEST">{props.children}</MainContent>
        </StyledLayoutContent>
      </Layout>
    </Container>
  );
};
