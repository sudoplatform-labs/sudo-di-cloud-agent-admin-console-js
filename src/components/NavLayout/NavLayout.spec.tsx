import { shallow } from 'enzyme';
import * as React from 'react';
import reactRouter from 'react-router';
import * as ConsoleRoutes from '../../containers/Console/routes';
import { NavLayout } from './NavLayout';

jest.spyOn(reactRouter, 'useLocation').mockReturnValue({
  location: {
    pathname: '/console',
  },
} as any);

jest.spyOn(ConsoleRoutes, 'getConsoleRoutesRoot').mockReturnValue({
  path: '/console',
});

describe('NavLayout', () => {
  it('should render', () => {
    const component = shallow(<NavLayout />);
    expect(component).toMatchSnapshot();
  });
});
