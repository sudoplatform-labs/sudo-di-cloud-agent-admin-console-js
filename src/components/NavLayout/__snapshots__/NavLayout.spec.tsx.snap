// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLayout should render 1`] = `
<Styled(Layout)>
  <Styled(Sider)
    width="252px"
  >
    <Sidebar
      routes={Array []}
    />
  </Styled(Sider)>
  <Layout>
    <Styled(Header)>
      <AppHeader />
    </Styled(Header)>
    <Styled(Content)
      id="STYLED_LAYOUT_CONTENT"
    >
      <BreadCrumbs
        routes={
          Array [
            Object {
              "path": "/console",
            },
          ]
        }
      />
      <styled.div
        id="TEST"
      />
    </Styled(Content)>
  </Layout>
</Styled(Layout)>
`;
